bayesian_cnn_prometheus/preprocessing/data_generator.py

```diff
@@ -0,0 +1,86 @@
+import functools
+from typing import Dict, Generator
+
+import numpy as np
+
+from bayesian_cnn_prometheus.constants import DatasetType
+from bayesian_cnn_prometheus.preprocessing.data_splitter import DataSplitter
+from bayesian_cnn_prometheus.preprocessing.image_loader import ImageLoader
+
+
+class DataGenerator:
+    def __init__(self, preprocessing_config: Dict, batch_size: int):
+        """
+        Creates PreprocessingPipeline class to preprocess input data according to the config.
+        :param preprocessing_config: structure with configuration to use in preprocessing in learning
+        """
+        self.batch_size = batch_size
+        self.should_normalise = preprocessing_config.get("normalize_images").get("is_activated")
+        self.image_loader = ImageLoader(preprocessing_config.get('transform_nifti_to_npy').get('ext'))
+        # TODO ProxPxD: is there a need for the activation of chunking?
+        self.chunk_size = preprocessing_config.get('create_chunks').get('chunk_size')
+        self.data_splitter = DataSplitter(preprocessing_config.get('create_data_structure'),
+                                          preprocessing_config.get('update_healthy_patients_indices'))
+        self.dataset_structure = None
+
+    def get_train(self):
+        return self._get_data_generator(DatasetType.TRAIN, self.batch_size)
+
+    def get_test(self):
+        return self._get_data_generator(DatasetType.TEST, self.batch_size)
+
+    def get_valid(self):
+        return self._get_data_generator(DatasetType.VALID, 1)
```

Comment:
Wygląda na to, że tutaj też powinna być wielkość taka jak w batchu, w sensie self.batch_size zamiast 1 :)